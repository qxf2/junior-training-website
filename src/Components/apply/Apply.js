import React, { Component } from "react";
import validator from "validator";
import emailjs, { init } from "emailjs-com";
import axios from "axios";
import DatePicker from "react-datepicker";
import Collapsible from "react-collapsible";
import "react-datepicker/dist/react-datepicker.css";

import { Modal } from "antd";
import "antd/dist/antd.css";

var rand = Math.random().toString(36).split(".")[1].slice(0, 8) + "";
var year = new Date().getFullYear();

let URL_Interview = "https://jsonplaceholder.typicode.com/users";

// Qxf2 EMAILJS credentials
let service_id = "default_service";
let template_id = "junior_training";
let user_id = "user_8k6FfpYGOYlfflIsrTFyF";

init(user_id);

class Apply extends Component {
  constructor(props) {
    super(props);

    // this state variable is use to set initial values for the required for the form
    this.state = {
      name: "",
      email: "",
      educationCollege: "",
      educationCourse: "",
      educationBranch: "",
      educationYear: new Date(),
      personalInfo: "",
      isCollegeProject: false,
      projects_outside: "",
      isHardware: "",
      git_url: "",
      video_software_url: "",
      video_url: "",
      linkedIn_url: "",
      about_testing: "",
      start_testing: "",
      programming_level: "",
      programming_language: "",
      how_u_find_us: "",
      technical_interest: "",
      looking_from_us: "",
      isEnabled: false,
    };

    // this is for the checkboxes at the bottom of the form
    this.checks = {
      check1: false,
      check2: false,
      check3: false,
      check4: false,
      check5: false,
    };

    // we can define all the  error variable here
    this.email_error = false;

    this.isSubmitted = false;

    //This variable is to define initial state which will be used once the form is submitted successfully
    this.baseState = this.state;

    // message for successfully submitting the application
    this.successMessage = "";

    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.onRadioChange = this.onRadioChange.bind(this);
    this.handleInputChange = this.handleInputChange.bind(this);
    this.handleSelectChange = this.handleSelectChange.bind(this);
    this.handleChangeSelect = this.handleChangeSelect.bind(this);
  }

  // function for handling the select options
  handleSelectChange = (programming_level) => {
    this.setState({ programming_level });
    this.setState({ isEnabled: this.setButton() });
  };

  // function for checkbox
  handleInputChange = (event) => {
    const id = event.target.id;

    switch (id) {
      case "check1":
        this.checks.check1 = !this.checks.check1;
        break;
      case "check2":
        this.checks.check2 = !this.checks.check2;
        break;
      case "check3":
        this.checks.check3 = !this.checks.check3;
        break;
      case "check4":
        this.checks.check4 = !this.checks.check4;
        break;
      case "check5":
        this.checks.check5 = !this.checks.check5;
        break;
      default:
        break;
    }
    this.setState({ isEnabled: this.setButton() });
  };

  // function to store all the input values according to their ID's
  handleChange = (event) => {
    const id = event.target.id;
    switch (id) {
      case "Name":
        if (event.target.value == "" || event.target.value.match(/^[A-Za-z ]+$/)) {
          this.setState({ name: event.target.value });
        }
        break;
      case "Email":
        this.setState({ email: event.target.value });
        this.email_error = true;
        if (validator.isEmail(event.target.value)|| validator.isEmpty(event.target.value)) {
          this.email_error = false;
        }
        break;
      case "education_college":
        this.setState({ educationCollege: event.target.value });
        break;
      case "education_course":
        this.setState({ educationCourse: event.target.value });
        break;
      case "education_branch":
        this.setState({ educationBranch: event.target.value });
        break;
      case "personalInfo":
        this.setState({ personalInfo: event.target.value });
        break;
      case "projects_outside":
        this.setState({ projects_outside: event.target.value });
        break;
      case "github":
        this.setState({ git_url: event.target.value });
        break;
      case "video_software":
        this.setState({ video_software_url: event.target.value });
        break;
      case "video":
        this.setState({ video_url: event.target.value });
        break;
      case "linkedIn":
        this.setState({ linkedIn_url: event.target.value });
        break;
      case "testing-knowledge":
        this.setState({ about_testing: event.target.value });
        break;
      case "software-testing":
        this.setState({ start_testing: event.target.value });
        break;
      case "languages":
        this.setState({ programming_language: event.target.value });
        break;
      case "find-us":
        this.setState({ how_u_find_us: event.target.value });
        break;
      case "looking-for":
        if (validator.isLength(event.target.value, { max: 100 })) {
          this.setState({ looking_from_us: event.target.value });
        }
        break;
      case "technical-interest":
        this.setState({ technical_interest: event.target.value });
        break;
      default:
        break;
    }
    this.setState({ isEnabled: this.setButton() });
  };

  // checking all validation from the users input
  setButton() {
    if (
      this.state.name.trim().length > 3 &&
      this.state.email.length > 3 &&
      this.state.educationCollege.trim().length > 3 &&
      this.state.educationCourse.trim().length >= 2 &&
      this.state.educationBranch.trim().length >= 2 &&
      this.state.educationYear != null &&
      this.state.personalInfo.trim().length > 3 &&
      this.state.linkedIn_url.trim().length > 3 &&
      this.state.start_testing.trim().length > 3 &&
      this.state.programming_language.trim().length > 3 &&
      this.state.programming_level != null &&
      this.state.how_u_find_us.trim().length > 3 &&
      this.state.technical_interest.trim().length > 3 &&
      this.state.looking_from_us.trim().length > 3 &&
      this.checks.check1 &&
      this.checks.check2 &&
      this.checks.check3 &&
      this.checks.check4 &&
      this.checks.check5
    ) {
      // this is for additional checks
      if (this.state.isCollegeProject) {
        if (
          (this.state.git_url.length >= 3 &&
            this.state.video_software_url.length >= 3) ||
          this.state.video_url.length >= 3
        ) {
          console.log("All set");
          return true;
        } else {
          return false;
        }
      } else {
        console.log("All set");
        return true;
      }
    } else {
      return false;
    }
  }

  // function to handle the radio options
  onRadioChange = (event) => {
    const name = event.target.name;
    const val = event.target.value;

    switch (name) {
      case "college-projects":
        if (val === "yes") {
          this.setState({ isCollegeProject: true });
        } else {
          this.setState({ isCollegeProject: false });
        }
        break;
      case "project-type":
        if (val === "hardware") {
          this.setState({ isHardware: true });
        } else {
          this.setState({ isHardware: false });
        }
        break;
      default:
        break;
    }
    this.setState({ isEnabled: this.setButton() });
  };

  handleChangeSelect = (event) => {
    this.setState({ programming_level: event.target.value });
    this.setState({ isEnabled: this.setButton() });
  };

  handleSubmit = (event) => {
    event.preventDefault();
    this.isSubmitted = true;
    this.setState({ isEnabled: this.setButton() });
    this.confirmSubmission();
  };

  // Function to send the email and for scheduling the interview
  confirmSubmission() {
    let Answer =
      "<b>Q1 Name</b> <br>" +
      this.state.name +
      "<br> <br> <b>Q2 Email</b> <br>" +
      this.state.email +
      "<br> <br> <b>Q3.a college </b> <br>" +
      this.state.educationCollege +
      "<br> <br> <b>Q3.b course </b> <br>" +
      this.state.educationCourse +
      "<br> <br> <b>Q3.c branch </b> <br>" +
      this.state.educationBranch +
      "<br> <br> <b>Q3.d year of pass out </b> <br>" +
      this.state.educationYear.getFullYear() +
      "<br> <br> <b>Q4 A little bit about yourself</b> <br>" +
      this.state.personalInfo +
      "<br> <br> <b>Q5 Have you done any projects outside of college?</b> <br>" +
      (this.state.isCollegeProject ? "Yes" : "No");

    if (this.state.isCollegeProject) {
      Answer +=
        "<br> <br> <b>Q5.1 What projects have you done outside of college?</b> <br>" +
        this.state.projects_outside +
        "<br> <br> <b>Q5.2 Hardware or software radio button</b> <br>" +
        (this.state.isHardware ? "Hardware" : "Software");

      if (this.state.isHardware) {
        Answer +=
          "<br> <br> <b>Q5.2.a (if hardware) Link to a Video of a working hardware project</b> <br>" +
          this.state.video_url;
      } else {
        Answer +=
          "<br> <br> <b>Q5.2.a (if software) GitHub profile</b> <br>" +
          this.state.git_url +
          "<br> <br> <b>Q5.2.b (if software) Link to a Video of a working software project</b> <br>" +
          this.state.video_software_url;
      }
    }

    Answer +=
      "<br> <br> <b>Q6 LinkedIn profile</b> <br>" +
      this.state.linkedIn_url +
      "<br> <br> <b>Q7 What do you know about testing?</b> <br>" +
      this.state.about_testing +
      "<br> <br> <b>Q8 Would you like to start your career in software testing? Why or why not?</b> <br>" +
      this.state.start_testing +
      "<br> <br> <b>Q9 Level of programming</b> <br>" +
      this.state.programming_level +
      "<br> <br> <b>Q9.1 What languages do you code in?</b> <br>" +
      this.state.programming_language +
      "<br> <br> <b>Q10 How did you find out about us?</b> <br>" +
      this.state.how_u_find_us +
      "<br> <br> <b>Q11 What are you looking for from this program?</b> <br>" +
      this.state.looking_from_us +
      "<br> <br> <b>Q12 What are your technical interests?</b> <br>" +
      this.state.technical_interest;

    this.successMessage =
      "Thank You for applying with us " +
      this.state.name +
      "! We will get back to you within 5 business days";

    let template_params = {
      contact_number: rand,
      answers: Answer,
    };

    if (this.setButton()) {
      // For emailjs all the credentials are defined on top of the file
      emailjs.send(service_id, template_id, template_params).then(
        (result) => {
          Modal.success({
            content: this.successMessage,
            onOk: async () => {
              window.location.reload(true);
            },
          });
        },
        (error) => {
          console.log(error.text);
        }
      );

      // For getting the API response
      axios.get(URL_Interview).then((res) => {
        if (res.status === 200) {
          const persons = res.data;
          // this.setState({ persons });
        }
      });
    } else {
      this.isSubmitted = false;
      Modal.error({
        content:
          "You need to fill all the req. fields(min 4 chars) in the application",
      });
    }
  }

  // once the app is loaded then this function is called
  componentDidMount() {
    console.log("Find a bug and the course price is free");
  }

  render() {
    const imp = <span className="required">*</span>;

    return (
      <section id="apply" data-testid="apply">
        <div className="row education">
          <div className="eight columns header-col">
            <h1>
              <span>Apply Here</span>
            </h1>
          </div>

          <div className="nine columns main-col">
            <div className="row item"></div>
          </div>
        </div>

        <div className="row work" data-testid="form">
          <div className="twelve columns header-col">
            <form onSubmit={this.handleSubmit}>
              <label>
                Name {imp}
                <input
                  data-testid="name"
                  id="Name"
                  className="input-text form-control form-control-lg"
                  type="text"
                  value={this.state.name}
                  ref={this.state.name}
                  onChange={this.handleChange}
                />
              </label>

              <label>
                Email {imp}
                <input
                  data-testid="email"
                  id="Email"
                  className="input-text"
                  type="text"
                  value={this.state.email}
                  ref={this.state.email}
                  onChange={this.handleChange}
                />
                {this.email_error ? (
                  <span className="required">Email not valid</span>
                ) : null}
              </label>

              <label>
                Enter all your education detail expanding the Education{imp}
              </label>

              <Collapsible trigger={<a href="">Education</a>}>
                <div
                  style={{
                    display: "flex",
                    flexDirection: "column",
                    padding: 15,
                  }}
                >
                  <label>College{imp}</label>
                  <input
                    id="education_college"
                    data-testid="college"
                    className="input-text"
                    type="text"
                    value={this.state.educationCollege}
                    ref={this.state.educationCollege}
                    onChange={this.handleChange}
                  />

                  <label>Course{imp}</label>
                  <input
                    id="education_course"
                    className="input-text"
                    data-testid="course"
                    type="text"
                    value={this.state.educationCourse}
                    onChange={this.handleChange}
                  />

                  <label>Branch{imp}</label>
                  <input
                    id="education_branch"
                    data-testid="branch"
                    className="input-text"
                    type="text"
                    value={this.state.educationBranch}
                    onChange={this.handleChange}
                  />

                  <label data-testid="year">year of pass out{imp}</label>

                  <DatePicker
                    placeholderText="Select Year"
                    id="education_year"
                    className="input-text"
                    selected={this.state.educationYear}
                    onChange={(date) => this.setState({ educationYear: date })}
                    showYearPicker
                    minDate={new Date(year - 1, 0, 1)}
                    maxDate={new Date(year + 1, 11, 31)}
                    dateFormat="yyyy"
                    yearItemNumber={11}
                  />

                  <span
                    style={{
                      fontWeight: "bold",
                      color: "red",
                    }}
                  >
                    {this.year_error}
                  </span>
                </div>
              </Collapsible>

              <label>
                A little bit about yourself {imp}
                <textarea
                  id="personalInfo"
                  className="input-text"
                  data-testid="personalInfo"
                  type="text"
                  value={this.state.personalInfo}
                  onChange={this.handleChange}
                />
              </label>

              <label>
                Have you done any projects outside of college? {imp}
                <br />
                <div onChange={this.onRadioChange}>
                  <label>
                    <input type="radio" value="yes" name="college-projects" />
                    &nbsp; Yes
                  </label>

                  <label>
                    <input type="radio" value="no" name="college-projects" />{" "}
                    &nbsp; No
                  </label>
                </div>
                <br />
              </label>
              {this.state.isCollegeProject && (
                <div>
                  <label>
                    What projects have you done outside of college? {imp}
                    <textarea
                      id="projects_outside"
                      className="input-text"
                      type="text"
                      value={this.state.projectOutside}
                      onChange={this.handleChange}
                    />
                  </label>

                  <label>
                    What is your project related to? {imp}
                    <br />
                    <div onChange={this.onRadioChange}>
                      <input
                        type="radio"
                        value="hardware"
                        name="project-type"
                      />
                      Hardware
                      <br />
                      <input
                        type="radio"
                        value="software"
                        name="project-type"
                      />
                      Software
                    </div>
                  </label>
                  {!this.state.isHardware ? (
                    <div>
                      <label>
                        GitHub profile {imp}
                        <input
                          id="github"
                          className="input-text"
                          type="text"
                          value={this.state.git_url}
                          onChange={this.handleChange}
                        />
                      </label>

                      <label>
                        Link to a Video of a working software project {imp}
                        <input
                          id="video_software"
                          className="input-text"
                          type="text"
                          value={this.state.video_software_url}
                          onChange={this.handleChange}
                        />
                      </label>
                    </div>
                  ) : (
                    <label>
                      Link to a Video of a working hardware project {imp}
                      <input
                        id="video"
                        className="input-text"
                        type="text"
                        value={this.state.video_url}
                        onChange={this.handleChange}
                      />
                    </label>
                  )}
                </div>
              )}

              <label>
                LinkedIn profile {imp}
                <input
                  id="linkedIn"
                  data-testid="linkedIn"
                  className="input-text"
                  type="text"
                  value={this.state.linkedIn_url}
                  onChange={this.handleChange}
                />
              </label>

              <label>
                What do you know about testing? {imp}
                <textarea
                  id="testing-knowledge"
                  data-testid="testing"
                  className="input-text"
                  type="text"
                  value={this.state.about_testing}
                  onChange={this.handleChange}
                />
              </label>

              <label>
                Would you like to start your career in software testing? Why or why not?{imp}
                <textarea
                  id="software-testing"
                  data-testid="likeToTesting"
                  className="input-text"
                  type="text"
                  value={this.state.start_testing}
                  onChange={this.handleChange}
                />
              </label>

              <label
                data-testid="programs"
                htmlfor="selectId"
                style={{ width: "fit-content" }}
              >
                <label htmlFor="programs">Level of programming {imp}</label>
                <select
                  id="selectId"
                  data-testid="select"
                  defaultValue="Select...."
                  onChange={this.handleChangeSelect}
                >
                  <option data-testid="select-option" value="beginner">
                    Beginner
                  </option>
                  <option data-testid="select-option" value="intermediate">
                    Intermediate
                  </option>
                  <option data-testid="select-option" value="expert">
                    Expert
                  </option>
                </select>
              </label>

              <label>
                What languages do you code in? {imp}
                <input
                  id="languages"
                  className="input-text"
                  data-testid="programmingLang"
                  type="text"
                  value={this.state.programming_language}
                  onChange={this.handleChange}
                />
              </label>

              <label>
                How did you find out about us? {imp}
                <input
                  id="find-us"
                  data-testid="findUs"
                  className="input-text"
                  type="text"
                  value={this.state.how_u_find_us}
                  onChange={this.handleChange}
                />
              </label>

              <label>
                What are you looking for from this program? (max 100 chars)
                {imp}
                <input
                  id="looking-for"
                  data-testid="lookingFor"
                  className="input-text"
                  type="text"
                  value={this.state.looking_from_us}
                  onChange={this.handleChange}
                />
              </label>

              <label>
                What are your technical interests? {imp}
                <input
                  id="technical-interest"
                  data-testid="techInterest"
                  className="input-text"
                  type="text"
                  value={this.state.technical_interest}
                  onChange={this.handleChange}
                />
              </label>

              <label>Do you understand that <small><a href="/#before-you-apply">[learn more]</a></small>:</label>

              <div>
                <label className="container" data-testid="check1">
                  fully remote{" "}
                  <input
                    type="checkbox"
                    id="check1"
                    onChange={this.handleInputChange}
                  />{" "}
                  {imp}
                  <span className="checkmark"></span>
                </label>

                <label className="container" data-testid="check2">
                  {" "}
                  not for everyone{" "}
                  <input
                    type="checkbox"
                    id="check2"
                    onChange={this.handleInputChange}
                  />{" "}
                  {imp}
                  <span className="checkmark"></span>
                </label>

                <label className="container" data-testid="check3">
                  {" "}
                  fast paced{" "}
                  <input
                    type="checkbox"
                    id="check3"
                    onChange={this.handleInputChange}
                  />{" "}
                  {imp}
                  <span className="checkmark"></span>
                </label>

                <label className="container" data-testid="check4">
                  {" "}
                  no placement guarantee{" "}
                  <input
                    type="checkbox"
                    id="check4"
                    onChange={this.handleInputChange}
                  />{" "}
                  {imp}
                  <span className="checkmark"></span>
                </label>

                <label className="container" data-testid="check5">
                  {" "}
                  no money back{" "}
                  <input
                    type="checkbox"
                    id="check5"
                    onChange={this.handleInputChange}
                  />{" "}
                  {imp}
                  <span className="checkmark"></span>
                </label>
              </div>
              {!this.state.isEnabled ? (
                <span className="required">
                  please fill all the required* fields(min 4 characters each) to
                  submit your application <br />{" "}
                </span>
              ) : null}

              {this.isSubmitted ? (
                <span id="image-loader">
                  <img alt="" src="images/loader.gif" />
                </span>
              ) : (
                <input
                  style={{
                    borderRadius: 5,
                    marginTop: 10,
                    width: 100,
                    padding: 4,
                  }}
                  className={!this.state.isEnabled ? "disabled" : null}
                  disabled={!this.state.isEnabled}
                  data-testid="submit"
                  type="submit"
                  value="submit"
                />
              )}
            </form>
          </div>
        </div>
      </section>
    );
  }
}

export default Apply;
