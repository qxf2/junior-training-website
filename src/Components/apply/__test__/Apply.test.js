import React from "react";
import ReactDOM from "react-dom";
import Apply from "../Apply";

import { render, cleanup, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

import renderer from "react-test-renderer";

afterEach(cleanup);

const inputProps = [
  { testId: "email", val: "testing@gmail.com" },
  { testId: "name", val: "Sailesh" },
  { testId: "college", val: "collegeName" },
  { testId: "course", val: "MS" },
  { testId: "branch", val: "EEE" },
  // { testId: "year", val : new Date() },
  { testId: "personalInfo", val: "I am a developer" },
  // college project
  { testId: "linkedIn", val: "some URl" },
  { testId: "testing", val: "none" },
  { testId: "likeToTesting", val: "yes I would" },
  // for level of languages
  { testId: "programmingLang", val: "JAVA etc." },
  { testId: "findUs", val: "online" },
  { testId: "lookingFor", val: "testing exp" },
  { testId: "techInterest", val: "some interest" },
];

const checkboxProps = [
  { text: "is fully remote", testId: "check1" },
  { text: "is not for everyone", testId: "check2" },
  { text: "is fast paced", testId: "check3" },
  { text: "has no placement guarantee", testId: "check4" },
  { text: "has no refund policy", testId: "check5" },
];

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Apply></Apply>, div);
});

it("renders apply correctly", () => {
  const { getByTestId } = render(<Apply></Apply>);
  expect(getByTestId("apply"));
});

it("Apply component should have name with the id of Name", () => {
  const { getByTestId } = render(<Apply />);
  const inputNameNode = getByTestId("name");
  expect(inputNameNode.id).toBe("Name");
});

it("Apply component should have email", () => {
  const { getByTestId } = render(<Apply />);
  const inputEmailNode = getByTestId("email");

  expect(inputEmailNode).toBeInTheDocument();
});

/*it("Email input should accept text", () => {
  const { getByTestId, getByText } = render(<Apply />);
  const inputEmailNode = getByTestId("email");

  expect(inputEmailNode.value).toMatch("");

  fireEvent.change(inputEmailNode, { target: { value: "testing" } });

  expect(inputEmailNode.value).toMatch("testing");

  const errorMsgNode = getByText("Email not valid");

  // to check the email validation fails
  expect(errorMsgNode).toBeInTheDocument();

  fireEvent.change(inputEmailNode, { target: { value: "testing@domain.com" } });

  // to check the email validation passes
  expect(errorMsgNode).not.toBeInTheDocument();
});*/

it("Should be able to submit the form", () => {
  const mockFn = jest.fn();
  const { getByRole } = render(<Apply handleSubmit={mockFn} />);
  const buttonNode = getByRole("button");
  fireEvent.submit(buttonNode);

  expect(mockFn).toHaveBeenCalledTimes(0);
});

it("Test", async () => {
  const mockFn = jest.fn();
  const { getByTestId, getByText, getByLabelText, getAllByTestId, getByRole } =
    render(<Apply onSubmit={mockFn} />);

  const errorMsgNode = getByText(
    "please fill all the required* fields(min 4 characters each) to submit your application"
  );

  inputProps.forEach((node) => {
    const inputNode = getByTestId(node.testId);
    fireEvent.change(inputNode, { target: { value: node.val } });
  });

  // to check the email validation passes
  expect(errorMsgNode).toBeInTheDocument();

  checkboxProps.forEach((node) => {
    const inputNode = getByText(node.text);
    fireEvent.click(inputNode);
    expect(getByTestId(node.testId).querySelector("input").checked).toBe(true);
  });

  const radio = getByLabelText("Yes");
  const radioSecond = getByLabelText("No");
  fireEvent.click(radioSecond);
  // check if false
  expect(radioSecond).toHaveProperty("checked", true);

  //The value should be the key of the option
  fireEvent.change(getByTestId("select"), { target: { value: "beginner" } });
  let options = getAllByTestId("select-option");
  expect(options[0].selected).toBeTruthy();
  expect(options[1].selected).toBeFalsy();
  expect(options[2].selected).toBeFalsy();

  // to check the email validation passes
  expect(errorMsgNode).not.toBeInTheDocument();

  const buttonNode = getByTestId("submit");
  fireEvent.submit(buttonNode);

  console.log(fireEvent.submit(buttonNode));

  // expect(mockFn).toHaveBeenCalled();
  // expect(mockFn).toHaveBeenCalledTimes(1);
});

it("Apply.js matches snapshot", () => {
  const tree = renderer.create(<Apply />).toJSON();
  expect(tree).toMatchSnapshot();
});