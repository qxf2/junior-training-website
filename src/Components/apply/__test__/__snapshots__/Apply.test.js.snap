// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Apply.js matches snapshot 1`] = `
<section
  data-testid="apply"
  id="apply"
>
  <div
    className="row education"
  >
    <div
      className="eight columns header-col"
    >
      <h1>
        <span>
          Apply Here
        </span>
      </h1>
    </div>
    <div
      className="nine columns main-col"
    >
      <div
        className="row item"
      />
    </div>
  </div>
  <div
    className="row work"
    data-testid="form"
  >
    <div
      className="twelve columns header-col"
    >
      <form
        onSubmit={[Function]}
      >
        <label>
          Name 
          <span
            className="required"
          >
            *
          </span>
          <input
            className="input-text form-control form-control-lg"
            data-testid="name"
            id="Name"
            onChange={[Function]}
            type="text"
            value=""
          />
        </label>
        <label>
          Email 
          <span
            className="required"
          >
            *
          </span>
          <input
            className="input-text"
            data-testid="email"
            id="Email"
            onChange={[Function]}
            type="text"
            value=""
          />
        </label>
        <label>
          Enter all your education detail expanding the Education
          <span
            className="required"
          >
            *
          </span>
        </label>
        <div
          className="Collapsible"
        >
          <span
            className="Collapsible__trigger is-closed"
            onClick={[Function]}
            onKeyPress={[Function]}
            style={null}
            tabIndex={null}
          >
            <a
              href=""
            >
              Education
            </a>
          </span>
          <div
            className="Collapsible__contentOuter"
            hidden={false}
            onTransitionEnd={[Function]}
            style={
              Object {
                "WebkitTransition": "height 400ms linear",
                "height": 0,
                "msTransition": "height 400ms linear",
                "overflow": "hidden",
                "transition": "height 400ms linear",
              }
            }
          >
            <div
              className="Collapsible__contentInner"
            >
              <div
                style={
                  Object {
                    "display": "flex",
                    "flexDirection": "column",
                    "padding": 15,
                  }
                }
              >
                <label>
                  College
                  <span
                    className="required"
                  >
                    *
                  </span>
                </label>
                <input
                  className="input-text"
                  data-testid="college"
                  id="education_college"
                  onChange={[Function]}
                  type="text"
                  value=""
                />
                <label>
                  Course
                  <span
                    className="required"
                  >
                    *
                  </span>
                </label>
                <input
                  className="input-text"
                  data-testid="course"
                  id="education_course"
                  onChange={[Function]}
                  type="text"
                  value=""
                />
                <label>
                  Branch
                  <span
                    className="required"
                  >
                    *
                  </span>
                </label>
                <input
                  className="input-text"
                  data-testid="branch"
                  id="education_branch"
                  onChange={[Function]}
                  type="text"
                  value=""
                />
                <label
                  data-testid="year"
                >
                  year of pass out
                  <span
                    className="required"
                  >
                    *
                  </span>
                </label>
                <div
                  className="react-datepicker-wrapper"
                >
                  <div
                    className="react-datepicker__input-container"
                  >
                    <input
                      className="input-text"
                      disabled={false}
                      id="education_year"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onClick={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      placeholder="Select Year"
                      readOnly={false}
                      type="text"
                      value="2021"
                    />
                  </div>
                </div>
                <span
                  style={
                    Object {
                      "color": "red",
                      "fontWeight": "bold",
                    }
                  }
                />
              </div>
            </div>
          </div>
        </div>
        <label>
          A little bit about yourself 
          <span
            className="required"
          >
            *
          </span>
          <textarea
            className="input-text"
            data-testid="personalInfo"
            id="personalInfo"
            onChange={[Function]}
            type="text"
            value=""
          />
        </label>
        <label>
          Have you done any projects outside of college? 
          <span
            className="required"
          >
            *
          </span>
          <br />
          <div
            onChange={[Function]}
          >
            <label>
              <input
                name="college-projects"
                type="radio"
                value="yes"
              />
                Yes
            </label>
            <label>
              <input
                name="college-projects"
                type="radio"
                value="no"
              />
               
                No
            </label>
          </div>
          <br />
        </label>
        <label>
          LinkedIn profile 
          <span
            className="required"
          >
            *
          </span>
          <input
            className="input-text"
            data-testid="linkedIn"
            id="linkedIn"
            onChange={[Function]}
            type="text"
            value=""
          />
        </label>
        <label>
          What do you know about testing? 
          <span
            className="required"
          >
            *
          </span>
          <textarea
            className="input-text"
            data-testid="testing"
            id="testing-knowledge"
            onChange={[Function]}
            type="text"
            value=""
          />
        </label>
        <label>
          Would you like to start your career in software testing? Why or why not?
          <span
            className="required"
          >
            *
          </span>
          <textarea
            className="input-text"
            data-testid="likeToTesting"
            id="software-testing"
            onChange={[Function]}
            type="text"
            value=""
          />
        </label>
        <label
          data-testid="programs"
          htmlfor="selectId"
          style={
            Object {
              "width": "fit-content",
            }
          }
        >
          <label
            htmlFor="programs"
          >
            Level of programming 
            <span
              className="required"
            >
              *
            </span>
          </label>
          <select
            data-testid="select"
            defaultValue="Select...."
            id="selectId"
            onChange={[Function]}
          >
            <option
              data-testid="select-option"
              value="beginner"
            >
              Beginner
            </option>
            <option
              data-testid="select-option"
              value="intermediate"
            >
              Intermediate
            </option>
            <option
              data-testid="select-option"
              value="expert"
            >
              Expert
            </option>
          </select>
        </label>
        <label>
          What languages do you code in? 
          <span
            className="required"
          >
            *
          </span>
          <input
            className="input-text"
            data-testid="programmingLang"
            id="languages"
            onChange={[Function]}
            type="text"
            value=""
          />
        </label>
        <label>
          How did you find out about us? 
          <span
            className="required"
          >
            *
          </span>
          <input
            className="input-text"
            data-testid="findUs"
            id="find-us"
            onChange={[Function]}
            type="text"
            value=""
          />
        </label>
        <label>
          What are you looking for from this program? (max 100 chars)
          <span
            className="required"
          >
            *
          </span>
          <input
            className="input-text"
            data-testid="lookingFor"
            id="looking-for"
            onChange={[Function]}
            type="text"
            value=""
          />
        </label>
        <label>
          What are your technical interests? 
          <span
            className="required"
          >
            *
          </span>
          <input
            className="input-text"
            data-testid="techInterest"
            id="technical-interest"
            onChange={[Function]}
            type="text"
            value=""
          />
        </label>
        <label>
          Do you understand that:
        </label>
        <div>
          <label
            className="container"
            data-testid="check1"
          >
            fully remote
             
            <input
              id="check1"
              onChange={[Function]}
              type="checkbox"
            />
             
            <span
              className="required"
            >
              *
            </span>
            <span
              className="checkmark"
            />
          </label>
          <label
            className="container"
            data-testid="check2"
          >
             
            not for everyone
             
            <input
              id="check2"
              onChange={[Function]}
              type="checkbox"
            />
             
            <span
              className="required"
            >
              *
            </span>
            <span
              className="checkmark"
            />
          </label>
          <label
            className="container"
            data-testid="check3"
          >
             
            fast paced
             
            <input
              id="check3"
              onChange={[Function]}
              type="checkbox"
            />
             
            <span
              className="required"
            >
              *
            </span>
            <span
              className="checkmark"
            />
          </label>
          <label
            className="container"
            data-testid="check4"
          >
             
            no placement guarantee
             
            <input
              id="check4"
              onChange={[Function]}
              type="checkbox"
            />
             
            <span
              className="required"
            >
              *
            </span>
            <span
              className="checkmark"
            />
          </label>
          <label
            className="container"
            data-testid="check5"
          >
             
            no money back
             
            <input
              id="check5"
              onChange={[Function]}
              type="checkbox"
            />
             
            <span
              className="required"
            >
              *
            </span>
            <span
              className="checkmark"
            />
          </label>
        </div>
        <span
          className="required"
        >
          please fill all the required* fields(min 4 characters each) to submit your application 
          <br />
           
        </span>
        <input
          className="disabled"
          data-testid="submit"
          disabled={true}
          style={
            Object {
              "borderRadius": 5,
              "marginTop": 10,
              "padding": 4,
              "width": 100,
            }
          }
          type="submit"
          value="submit"
        />
      </form>
    </div>
  </div>
</section>
`;
